### PR TITLE
Added zapProxy to zapTest script

### PR DESCRIPTION
**Pull request by thadeusssewanyana**

### CODE REVIEW

File: run_statement_of_liability_api_zap_tests.sh [hunk 1]
```diff
@@ -10,6 +12,6 @@ fi
 
 echo "*** running on $environment for tags '$tags' ***"
 
-sbt -Denvironment="$environment" -Dcucumber.options="--tags '$tags'" clean 'testOnly uk.gov.hmrc.test.api.cucumber.runner.StatementOfLiabilityApiTestRunner'
+sbt -Dhttp.proxyHost=localhost -Dhttp.proxyPort=11000 -Denvironment="$environment" -Dcucumber.options="--tags '$tags'" clean "testOnly uk.gov.hmrc.test.runners.StatementOfLiabilityApiTestRunner"
```
Comment (rogerball) [inline run_statement_of_liability_api_zap_tests.sh:15]: should this be uk.gov.hmrc.test.api.cucumber.runner ?

File: src/test/scala/uk/gov/hmrc/test/api/client/WsClient.scala [hunk 2]
```diff
@@ -17,19 +17,19 @@ object WsClient extends LazyLogging {
   implicit val bodyWrites: BodyWritable[JsValue] =
     BodyWritable(a => InMemoryBody(ByteString.fromArrayUnsafe(Json.toBytes(a))), "application/json")
   private val asyncClient: StandaloneAhcWSClient = {
-    implicit val system: ActorSystem = ActorSystem()
+    implicit val system: ActorSystem             = ActorSystem()
```
Comment (rogerball) [inline src/test/scala/uk/gov/hmrc/test/api/client/WsClient.scala:20]: Should we syncronise auto formatting?

File: src/test/scala/uk/gov/hmrc/test/api/client/WsClient.scala [hunk 3]
```diff
@@ -17,19 +17,19 @@ object WsClient extends LazyLogging {
...
-           headers: Map[String, String] = Map.empty[String, String],
-           cookies: Seq[WSCookie] = Seq.empty[WSCookie],
-           followRedirects: Boolean = false
-         ): StandaloneWSResponse = {
+    uri: String,
+    queryParameters: Map[String, String] = Map.empty[String, String],
+    headers: Map[String, String] = Map.empty[String, String],
+    cookies: Seq[WSCookie] = Seq.empty[WSCookie],
+    followRedirects: Boolean = false
+  ): StandaloneWSResponse = {
```
Comment (rogerball) [inline src/test/scala/uk/gov/hmrc/test/api/client/WsClient.scala:32]: Should we syncronise auto formatting?

File: run_interest_forecasting_api_tests.sh [hunk 4]
```diff
@@ -5,7 +5,7 @@ tags="not @smoke and not @ignore"
 if [ $# -gt 0 -a "$1" != "$environment" ];
 then
   environment="$1"
-  tags="not @ignore"
+  tags="@isoZapTest"
```
Comment (kiranpl) [inline run_interest_forecasting_api_tests.sh:8]: Is this intended? IFS tests limited to Zap tests? What does isoZapTest mean?

File: src/test/resources/application.conf [hunk 5]
```diff
@@ -46,17 +46,17 @@ interestForecastingApi {
...
+statementOfLiabilityApi {
   ignoreOptimizelyAlerts: true
   failureThreshold: "Medium"
-  testUrl: "http://localhost:0000/statement-of-liability"
-  contextBaseUrlRegex: "http://localhost:0000.*"
-  alertUrlsToReport: ["http://localhost:0000"]
+  testUrl: "http://localhost:9945/statement-of-liability/hello-world"
+  contextBaseUrlRegex: "http://localhost:9945.*"
+  alertUrlsToReport: ["http://localhost:9945"]
   testingAnApi = false
```
Comment (kiranpl) [inline src/test/resources/application.conf]: IFS Zap config has "testingAnApi" set to true and SoL Zap config has "testingAnApi" set to false?

File: src/test/scala/uk/gov/hmrc/test/api/client/WsClient.scala (hunk 3 above)
Comment (thadeusssewanyana) [reply src/test/scala/uk/gov/hmrc/test/api/client/WsClient.scala:32]: i think the repo standard is sbt test:scalafmt

File: run_interest_forecasting_api_tests.sh (hunk 4 above)
Comment (thadeusssewanyana) [reply run_interest_forecasting_api_tests.sh:8]: we don't need this tag now, is removed

File: src/test/resources/application.conf (hunk 5 above)
Comment (thadeusssewanyana) [reply src/test/resources/application.conf]: the test result is the same for both true or false but I have updated it to true incase it affects later testing

**Pull request closed without merging**